kernel/src/device/tdxguest/mod.rs

```diff
@@ -168,6 +177,42 @@ struct TdxQuoteHdr {
     out_len: u32,
 }
 
+/// TDX Report structure (`TDREPORT_STRUCT`) as defined in the Intel TDX Module Specification.
+#[repr(C)]
+struct TdReport {
+    report_mac: ReportMac,
+    _reserved: [u8; 256],
+    td_info: TdInfo,
+}
+
+#[repr(C)]
+struct ReportMac {
+    _reserved1: [u8; 128],
+    report_data: [u8; 64],
+    _reserved2: [u8; 64],
+}
+
+#[repr(C)]
+struct TdInfo {
+    attributes: [u8; 8],
+    xfam: [u8; 8],
+    mrtd: [u8; 48],
+    mrconfigid: [u8; 48],
+    mrowner: [u8; 48],
+    mrownerconfig: [u8; 48],
+    rtmr1: [u8; 48],
+    rtmr2: [u8; 48],
+    rtmr3: [u8; 48],
+    rtmr4: [u8; 48],
+    servtd_hash: [u8; 48],
+}
+
+impl TdReport {
+    fn report_data_offset() -> usize {
+        offset_of!(ReportMac, report_data) + offset_of!(TdReport, report_mac)
```

Comment:
First, make this function a const so that the value can be determined at the compile time.

Second, reverse the components in the sum expression to make it more natural in logic:

```rust
        offset_of!(TdReport, report_mac) + offset_of!(ReportMac, report_data)
```